inference/core/workflows/core_steps/sampling/identify_outliers/v1.py

```diff
@@ -0,0 +1,197 @@
+        examples=[100],
+    )
+
+    window_size: Optional[Union[Selector(kind=[INTEGER_KIND]), int]] = Field(
+        default=32,
+        description="The number of previous data points to consider in the sliding window algorithm.",
+        examples=[5],
+    )
+
+    @classmethod
+    def describe_outputs(cls) -> List[OutputDefinition]:
+        return [
+            OutputDefinition(name="is_outlier", kind=[BOOLEAN_KIND]),
+            OutputDefinition(name="percentile", kind=[FLOAT_ZERO_TO_ONE_KIND]),
+            OutputDefinition(name="warming_up", kind=[BOOLEAN_KIND]),
+        ]
+
+    @classmethod
+    def get_execution_engine_compatibility(cls) -> Optional[str]:
+        return ">=1.3.0,<2.0.0"
+
+
+class IdentifyOutliersBlockV1(WorkflowBlock):
+    def __init__(self):
+        self.samples = 0
+
+        # Keep track of all embeddings for vMF parameter estimation:
+        self.all_embeddings = []  # Store normalized embeddings
+
+    @classmethod
+    def get_manifest(cls) -> Type[WorkflowBlockManifest]:
+        return BlockManifest
+
+    def _fit_vmf_parameters(self, embeddings: np.ndarray):
+        """
+        Fit a von Mises-Fisher distribution to the given set of unit-normalized embeddings.
+        Returns:
+            mu (np.ndarray): Mean direction vector.
+            kappa (float): Concentration parameter.
+        """
+        n, d = embeddings.shape
+        if n < 2:
+            # Not enough data to fit
+            return None, None
+
+        # Sum all embeddings:
+        sum_vec = np.sum(embeddings, axis=0)
+        R = np.linalg.norm(sum_vec)
+        if R == 0:
+            # All embeddings canceled out, no direction
+            return None, None
+
+        mu = sum_vec / R
+        r_bar = R / n
+
+        # Approximate kappa:
+        # For d>2, a known approximation:
+        # kappa ≈ r_bar*(d - r_bar^2)/(1 - r_bar^2)
+        d = float(d)
+        if r_bar < 1.0:
+            kappa = (r_bar * (d - r_bar**2)) / (1 - r_bar**2)
+        else:
+            # Degenerate case: all points are identical
+            kappa = np.inf
+
+        return mu, kappa
+
+    def run(
+        self,
+        embedding: List[float],
+        threshold_percentile: float,
+        warmup: int,
+        window_size: int,
+    ) -> BlockResult:
+        # Convert to np array
+        embedding = np.array(embedding, dtype=float)
+        # Normalize embedding for vMF
+        norm = np.linalg.norm(embedding)
+        if norm == 0:
+            # If zero vector, skip normalization
+            embedding_normed = embedding
+        else:
+            embedding_normed = embedding / norm
+
+        self.samples += 1
+        warming_up = self.samples < warmup
+
+        # Store normalized embedding for vMF:
+        self.all_embeddings.append(embedding_normed)
+        if len(self.all_embeddings) > window_size:
+            # Remove oldest embedding
+            self.all_embeddings.pop(0)
```

Comment:
probably no a big deal for window sizes we are dealing with, but I think if we're popping from beginning, `deque` might be better/faster data structure than a list.